Automattic/jetpack
# VideoPress: implement a custom VideoPress video block recovery

This PR suggests a way to recover the Jetpack VideoPress core/video block instances, converting them to the new VideoPress Video block `v6` when the instances have validation issues.

Fixes https://github.com/Automattic/jetpack/issues/26021

#### Changes proposed in this Pull Request:

* VideoPress: implement a custom VideoPress video block recovery

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






The testing flow is to mimic an actual situation:

* User has Jetpack plugin active, with VideoPress supported.
* Edit a post
* Create a video block
* Upload a video
* Confirm it is a VideoPress video (take a look at the player, for instance)
* Confirm it's a current VideoPress implementation (core/video block)

<img width="911" alt="Screen Shot 2022-10-12 at 09 52 15" src="https://user-images.githubusercontent.com/77539/195347505-ce7a76c0-eb31-45ed-8ce1-29c4f4134c13.png">

* Now, let's deactivate the Jetpack plugin
* And activate VideoPress standalone plugin
* Confirm you see the VideoPress block recovery attempt

<img width="681" alt="image" src="https://user-images.githubusercontent.com/77539/195347707-65144e3d-f3ea-4f6a-b724-640da8c49edc.png">

* Confirm you can convert the invalidated core/video block to the new VideoPress block video by clicking on the `Attempt VideoPress Block Recovery` button

<img width="915" alt="Screen Shot 2022-10-12 at 09 55 10" src="https://user-images.githubusercontent.com/77539/195348131-9b51f4b6-4807-4066-a657-ab01c206db09.png">

* Confirm it's skippable. 


https://user-images.githubusercontent.com/77539/195349340-8ad9ff29-93f8-48f0-911e-31ab4cc910f8.mov
